aiidalab/registry/releases.py

```diff
@@ -30,15 +30,62 @@ def _split_release_line(url):
     return url, None
 
 
+def _get_tags(repo, rev, rev_selection):
+    # While the git rev-list command supports listing revisions for a
+    # single ref, in this context we only support rev selections for a
+    # range, not for individual refs.
+    if ".." not in rev_selection:
+        raise ValueError(
+            "The rev_selection '{rev_selection}' must specify a range, "
+            "that means must contain the range operator '..'."
+        )
+
+    # Incomplete revision selections such as `@main:v1..` must be expanded to
+    # `@main:v1..main`. Therefore, we first determine the branch ref for the
+    # given rev:
+    for ref in [f"refs/heads/{rev}", f"refs/remotes/origin/{rev}"]:
+        if ref.encode() in repo.refs:
+            break
+    else:
+        raise RuntimeError(f"Revision '{rev}' not a valid branch name.")
+
+    # Transform a potentially incomplete rev_selection into one that
+    # contains the branch ref. For example `v1..` is expanded to
+    # `v1..{ref}`, where `{ref}` is replaced with the actual reference.
+    start, _, stop = rev_selection.rpartition("..")
+    selected_commits = repo.rev_list(f"{start or ref}..{stop or ref}")
+
+    for tag in repo.get_merged_tags(rev):
+        commit = repo.get_commit_for_tag(tag)
+        if commit in selected_commits:
+            yield tag, commit
+
+
 def _get_release_commits(repo, release_line):
+    """Get the commits for a release line."""
     match = re.match(RELEASE_LINE_PATTERN, release_line)
 
     if not match:
         raise ValueError(f"Invalid release line specification: {release_line}")
 
     rev = match.groupdict()["rev"] or repo.get_current_branch()
 
-    if match.groupdict()["rev_selection"] is None:
+    if match.groupdict()["rev"] == "*":
+        # loop over all remote branches and yield the tags for the commits
```

Comment:
Just wondering, is this potentially an expensive operation to iterate over all branches? Hopefully not, but it would be good to somehow test / verify.

Reply:
It is not, it runs pretty fast. I run `build.sh` in aiidalab-regstry locally and I cannot feel any delay compared to only fetching on one branch. In QeApp, we have quite a lot branches (this is not good 🥲 ) but it just quickly finished without I noticing it.